calculadora/src/components/calculadora.js

```diff
@@ -23,113 +24,280 @@ const theme = createTheme({
 });
 
 export default function Calculadora() {
+  const [num, setNum] = useState(0);
+  const [oldnum, setOldnum] = useState();
```

Comment:
É sempre uma boa prática deixar o valor padrão no useState()

```suggestion
  const [oldnum, setOldnum] = useState(0);
```

ou

```suggestion
  const [oldnum, setOldnum] = useState("0");
```

(Depende da lógica que você for adotar)

Reply:
0 ele é inteiro, "0" ele é uma string